stdlib/public/core/Sort.swift.gyb

```diff
@@ -190,7 +237,7 @@ func _introSortImpl<C>(
     depthLimit: depthLimit &- 1)
   _introSortImpl(
     &elements,
-    subRange: (elements.index(after: partIdx))..<range.upperBound,
+    subRange: partIdx..<range.upperBound,
```

Comment:
It's an intentional change. Because the previous implementation of `_partition` slotted the pivot element into `partIdx` as its final step, the elements were always arranged such that everything in `range.lowerBound..<partIndex` was less than `elements[partIdx]` and in `elements.index(after: partIdx)..<range.upperBound` was greater than or equal to `elements[partIdx]`. 

The new version doesn't have that guarantee, only that everything in `range.lowerBound..<partIdx` is less than everything in `partIdx..<range.upperBound`, so the sort of the upper partition needs to include the partitioning index.